luna/featurevis/featurevis.py

```diff
@@ -24,62 +24,124 @@ def __init__(self, iterations, learning_rate):
         self.iterations = iterations
         self.learning_rate = learning_rate
 
-
-class AugmentationParameters():
-    """Object for generalizing augmentation parameters.
+class AuxiliaryTransformationParameters:
+    """Object for generalizing auxiliary augmentation parameters.
 
     Args:
-        blur (bool): whether or not blur is applied.
-        scale (bool): whether or not scale is applied.
+        add_blur (bool): whether or not gaussian blur is applied.
+        scale (bool): whether or not scaling is applied.
         pad_crop (bool): whether or not random pad or crop are applied.
-        flip(bool): whether or not flip is applied.
-        rotation(bool): whether or not rotation is applied.
-        noise (bool): whether or not noise is applied.
+        add_flip(bool): whether or not flip is applied.
+        add_rotation(bool): whether or not vert_rotation is applied.
+        add_noise (bool): whether or not noise is applied.
         color_aug(bool): whether or not color augmentation is applied.
     """
 
-    def __init__(self, blur, scale, pad_crop, flip, rotation, noise, color_aug):
-        self.blur = blur
+    def __init__(
+        self,
+        add_blur=False,
+        scale=False,
+        pad_crop=False,
+        add_flip=False,
+        add_rotation=False,
+        add_noise=False,
+        color_aug=False,
+    ) -> None:
+        self.add_blur = add_blur
         self.scale = scale
         self.pad_crop = pad_crop
-        self.flip = flip
-        self.rotation = rotation
-        self.noise = noise
+        self.add_flip = add_flip
+        self.add_rotation = add_rotation
+        self.add_noise = add_noise
         self.color_aug = color_aug
 
+
+class TransformationParameters:
+    """Object for generalizing augmentation parameters.
+
+    Args:
+        pad_size (float): the amount of padding to be applied.
+        pad_mode (str): type of padding to be applied.
+        add_jitter (float): the amount of jitter to be applied.
+        bilinear (list): the amount of bilinear scaling to be applied.
+        rotation (list): the amount of rotation to be applied.
+    """
+
+    def __init__(
+        self,
+        pad_size=None,
+        pad_mode="REFLECT",
+        add_jitter=None,
+        bilinear=None,
+        rotation=None,
+    ) -> None:
+        self.pad_size = pad_size
+        self.pad_mode = pad_mode
+        self.add_jitter = add_jitter
+        self.rescale_val = bilinear
+        self.angles = rotation
+
 #pylint: disable=too-many-locals
 
 
-def visualize_filter(image, model, layer, filter_index, opt_param, aug_param):
+def visualize_filter(
+    image,
+    model,
+    layer,
+    filter_index,
+    opt_param,
+    trans_param=None,
+    aux_trans_param=None,
+    custom_trans=None,
```

Comment:
Why do we have 3 different ones? Wouldn't one be enough?

Reply:
Removed after functionalization of the transformations